schematic/schemas/data_model_graph.py

```diff
@@ -226,6 +229,67 @@ def get_adjacent_nodes_by_relationship(
 
         return list(nodes)
 
+    def get_component_node_validation_rules(
+        self, manifest_component: str, node_display_name: str
+    ) -> list:
+        """Get valdation rules for a given node and component.
+        Args:
+            manifest_component, str: manifest component display name that the node belongs to.
+            node_display_name, str: node display name for the node being queried.
+        Returns:
+            validation_rules, list[str]: validation rules list for a given node and component.
+        """
+        # get any additional validation rules associated with this node (e.g. can this node
+        # be mapped to a list of other nodes)
+        node_validation_rules = self.get_node_validation_rules(
+            node_display_name=node_display_name
+        )
+
+        # Parse the validation rules per component if applicable
+        if node_validation_rules and isinstance(node_validation_rules, dict):
+            node_validation_rules = extract_component_validation_rules(
+                manifest_component=manifest_component,
+                validation_rules_dict=node_validation_rules,
+            )
+        return node_validation_rules
+
+    def get_component_node_required(
```

Comment:
I think it will be worthwhile writing a unit test for get_component_node_required function? If this can't be added to this PR, maybe we could create a separate ticket?

I also noticed that `get_json_validation_schema` function calls: `get_component_node_validation_rules` function before calling get_component_node_required function. But get_component_node_required function calls `get_component_node_validation_rules`  function again. Is this intended? I think these are duplicated? And technically, `get_component_node_required` could just get output from `get_component_node_validation_rules`?

Reply:
Based on discussion with Mialy over slack, this "duplication" is expected because we want get_component_node_required to be called independent of get_component_node_validation_rules in the future.

Reply:
I actually added the ability to optionally provide the component validation rules, and also added them in the JSON Schema code.